tests/php/sync/test_class.jetpack-sync-callables.php

```diff
@@ -190,6 +190,44 @@ function test_sync_always_sync_changes_to_home_siteurl_right_away() {
 		update_option( 'siteurl', $original_siteurl_option );
 	}
 
+	function test_sync_jetpack_sync_unlock_sync_callable_action_allows_syncing_siteurl_changes() {
+		delete_transient( Jetpack_Sync_Module_Callables::CALLABLES_AWAIT_TRANSIENT_NAME );
+		delete_option( Jetpack_Sync_Module_Callables::CALLABLES_CHECKSUM_OPTION_NAME );
+		$this->setSyncClientDefaults();
+
+		$original_home_option    = get_option( 'home' );
+		$original_siteurl_option = get_option( 'siteurl' );
+
+		// Let's see if the original values get synced. This will also set the await transient.
+		$this->sender->do_sync();
+		$synced_home_url = $synced_value = $this->server_replica_storage->get_callable( 'home_url' );
+		$synced_site_url   = $synced_value = $this->server_replica_storage->get_callable( 'site_url' );
+
+		$this->assertEquals( $original_home_option, $synced_home_url );
+		$this->assertEquals( $original_siteurl_option, $synced_site_url );
+
+		$this->server_replica_storage->reset();
+
+		// We set the filters here to simulate how setting the WP_HOME and WP_SITEURL constant works.
+		add_filter( 'option_home',    array( $this, 'return_https_site_com_blog' ) );
+		add_filter( 'option_siteurl', array( $this, 'return_https_site_com_blog' ) );
+
+		// By calling this action, we simulate wp_schedule_single_event()
+		do_action( 'jetpack_sync_unlock_sync_callable' );
```

Comment:
Do we need it since we only call `do_action( 'jetpack_sync_unlock_sync_callable' );` in this test file?

Ideally, a developer would use something like `wp_schedule_single_event( time() + 1, 'jetpack_sync_unlock_sync_callable' )` to trigger clearing the callables.